infra/modules/domain/data/main.tf

```diff
@@ -0,0 +1,13 @@
+locals {
```

Comment:
Per a previous convo, the domain stuff for notifications should go in here too. Doesn't have to be in this PR though

Reply:
Could you clarify which stuff you are referring to? I thought I already refactored the domain stuff for notifications in this PR.

Are you referring to domain_identity? If so I don't think that belongs here. Even though it's called domain_identity it's really an email identity that's scoped to the entire domain, rather than a configuration of the domain itself.